src/render.rs

```diff
@@ -139,10 +139,13 @@ impl<'a> Default for MarkdownRenderer<'a> {
 /// use render::markdown_to_html;
 ///
 /// let text = "[Rust](https://rust-lang.org/) is an awesome *systems programming* language!";
-/// let rendered = markdown_to_html(text)?;
+/// let rendered = markdown_to_html(text, None)?;
 /// ```
-pub fn markdown_to_html(text: &str) -> CargoResult<String> {
-    let renderer = MarkdownRenderer::new();
+pub fn markdown_to_html(text: &str, repo: Option<&str>) -> CargoResult<String> {
+    let repo = repo.map(|r|
+        format!("{}{}blob/master/", r, if r.ends_with("/") { "" } else { "/" }));
```

Comment:
We already depend on the `url` crate, should we use it to check that the given `repo` url's scheme is either `http` or `https`? I think it is possible for someone to submit a crate with a repository link using the `git://...` scheme.

Also, would it be worth it to check that the hosts are either `github.com`, `gitlab.com` or `bitbucket.org` ?

Reply:
> should we use it to check that the given `repo` url's scheme is either `http` or `https`?

I've just looked into it; I think we're safe — the API already checks to ensure a `repository` has either `http` or `https`, and by the documentation [on crates.io](http://doc.crates.io/manifest.html#package-metadata) this is expected:

> These are intended to be webviews of the relevant data, not necessarily compatible with VCS tools and the like.

Here's the relevant code:

https://github.com/rust-lang/crates.io/blob/ef1ac581f9c4ea6a906347d43776418acfdbb8ca/src/krate/mod.rs#L183-L219

I think it's a good idea to check the host; in case it doesn't match any, I feel like we should just drop relative links.